### PR TITLE
Updates for Python 3.11 compatibility

### PR DESCRIPTION
**Pull request by Silarn**
Some of this may also just be general cleanup for recent changes

### CODE REVIEW

**Comment by Holt59** [comment]
Is this still needed?

**Comment by Silarn** [comment]
I think so, at least this all seems to be fixed for updates to pyqt6 and the differences in the pyi files as well as the necessary dependency version update for 3.11.

File: basic_features/basic_save_game_info.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def allFiles(self) -> list[str]:
 
 
 class BasicGameSaveGameInfoWidget(mobase.ISaveGameInfoWidget):
-    def __init__(self, parent: QWidget, get_preview: Callable[[Path], Path | None]):
+    def __init__(self, parent: QWidget, get_preview: Optional[Callable[[Path], Optional[QPixmap | Path | str]]]):
```
Comment (Holt59) [inline basic_features/basic_save_game_info.py:36]: `Optional` should not be used anymore, you can use `| None` when needed. I don't think making `get_preview` optional is needed, simply default it to a `lambda p: None`. I thought this was already done in an existing PR 🤔

File: basic_features/basic_save_game_info.py [hunk 2]
```diff
@@ -89,9 +89,9 @@ def setSave(self, save: mobase.ISaveGame):
 
 
 class BasicGameSaveGameInfo(mobase.SaveGameInfo):
-    def __init__(self, get_preview: Callable[[Path], Path | None] | None = None):
+    def __init__(self, get_preview: Optional[Callable[[Path], Optional[QPixmap | Path | str]]] = None):
```
Comment (Holt59) [inline basic_features/basic_save_game_info.py:92]: Same comment as above for `Optional`.

File: basic_features/basic_save_game_info.py [hunk 3]
```diff
@@ -89,9 +89,9 @@ def setSave(self, save: mobase.ISaveGame):
...
         super().__init__()
-        self._get_preview = get_preview
+        self._get_preview: Optional[Callable[[Path], Optional[QPixmap | Path | str]]] = get_preview
```
Comment (Holt59) [inline basic_features/basic_save_game_info.py:94]: No need to type here since `get_preview` (the parameter) is typed.

**Review by Holt59 — CHANGES_REQUESTED** [review]
Ok for me except for the `Optional` stuff that should use the new `| None` syntax.

**Comment by Holt59** [comment]
Also, CI will probably fail for `| None` because the Python version needs to be updated to 3.11 in the workflow file.

**Comment by Silarn** [comment]
I made the changes because my IDE was complaining that None was not supposed to be valid. I'll have to do this in a few hours.

**Comment by Holt59** [comment]
> I made the changes because my IDE was complaining that None was not supposed to be valid. I'll have to do this in a few hours.

I don't know what IDE you're using, but it's probably using a Python version lower than 3.10.

**Review by Holt59 — APPROVED** [review]

**Pull request merged**